includes/functions/payments.php

```
@@ -173,131 +34,82 @@ function awpcp_payfast_verify_received_data( $data = array() ) {
-
-    curl_close( $ch );
-
-    if ( in_array( $response, array( 'VALID', 'INVALID' ) ) ) {
-        $response = $response;
-    } else {
-        $response = 'ERROR';
-    }
+    $url = 'https://' . $host . '/eng/query/validate';
+
+    $args = array(
+        'method'      => 'POST',
+        'timeout'     => 30,
+        'redirection' => 5,
+        'httpversion' => '1.0',
+        'blocking'    => true,
+        'headers'     => array(
+            'Content-Type' => 'application/x-www-form-urlencoded',
+            'User-Agent'   => 'Another WordPress Classifieds Plugin',
+        ),
+        'body'        => $content,
+        'cookies'     => array(),
+        'sslverify'   => true,
+    );
 
-    return $response;
-}
+    $response = wp_remote_post( $url, $args );
 
-/**
- * @since 3.7.8
- */
-function awpcp_payfast_verify_received_data_with_fsockopen( $content ) {
-    if ( get_awpcp_option( 'paylivetestmode' ) ) {
-        $host = 'sandbox.payfast.co.za';
-    } else {
-        $host = 'www.payfast.co.za';
+    if ( is_wp_error( $response ) ) {
+        return 'ERROR';
     }
 
-    $header_processed = false;
-    $response = '';
-
-    $header = "POST /eng/query/validate HTTP/1.0\r\n";
-    $header .= 'Host: ' . $host . "\r\n";
-    $header .= "User-Agent: Another WordPress Classifieds Plugin\r\n";
-    $header .= "Content-Type: application/x-www-form-urlencoded\r\n";
-    $header .= 'Content-Length: ' . strlen( $content ) . "\r\n\r\n";
-
-    $socket = fsockopen( 'ssl://' . $host, 443, $errno, $errstr, 30 );
-
-    fputs( $socket, $header . $content );
-
-    while( ! feof( $socket ) ) {
-        $line = fgets( $socket, 1024 );
-
-        if ( strcmp( $line, "\r\n" ) == 0 ) {
-            $header_processed = true;
-        } elseif ( $header_processed ) {
-            $response .= $line;
-        }
+    $response_code = wp_remote_retrieve_response_code( $response );
+    if ( 200 !== $response_code ) {
+        return 'ERROR';
```

Instead of hardcoding `'ERROR'`, could you return an error like this?

```php
return new WP_Error(
    'no_response',
    __( 'Non-200 status code returned.', 'another-wordpress-classifieds-plugin' ),
    [ 'status' => $response_code ]
);
```